### DISCUSSION TITLE
useFieldArray remove/append method bugged?

### DISCUSSION THREAD

**Discussion by jasonlex** [react-hook-form/react-hook-form#6092] [Q&A] [answered]
For some reason I have a particular form setup with useFieldArray but the remove/append method is bugged. I have not encounter this problem with other forms using useFieldArray so I am not sure if it is an isolated case.

I have a sandbox link showing the problem with remove an item in the useFieldArray:

https://codesandbox.io/s/file-array-forked-5msnc?file=/src/index.js

**Step of reproducing the bug:**
1. Append a few arrays
2. Remove an array
3. Append again and you will see in the console that all other field arrays beside the new one becomes _null_ for some reason.

I am using react-hook-form 6.13.1 since the new versions breaks a number of stuff in my existing forms so I do not plan to upgrade the version at the moment...

How can I fix this?

**Comment by bluebill1049** [comment, marked as answer]
> It's important to apply ref={register()} instead of ref={register} when working with useFieldArray so register will get invoked during map.

https://react-hook-form.com/v6/api#useFieldArray

<img width="976" alt="Screen Shot 2021-07-27 at 8 00 13 pm" src="https://user-images.githubusercontent.com/10513364/127135504-1ccc30f8-fe4c-49df-b180-5c1260097e64.png">

`inputRef={register()}`

**Reply by jasonlex** [reply]
Thanks. That solved the problem. Will take note on that.
